sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -316,6 +317,21 @@ trait Row extends Serializable {
   }
 
   /**
+   * Returns the Optional value at position i.
+   * If the value is null or if it cannot be cast to the requested type, None is returned.
+   */
+  def getAsOpt[T](i: Int): Option[T] = {
```

Comment:
`getOption` would be more inline with similar functions in the Scala collections API (i.e. `headOption` `reduceOption`).  We also typically avoid abbreviation unless its very common.